### PR TITLE
Set resource limits for connector definitions: api layer

### PR DESCRIPTION
**Pull request by cgardens**
relates to: https://github.com/airbytehq/airbyte/issues/10295

## What
* Exposes the ability to set resource requirements for a connector definition in the API

## To do
- [x] - sanity check that this is going to work okay for seeded definitions.

## Recommended reading order
1. `x.java`
2. `y.python`

### CODE REVIEW

File: airbyte-config/init/src/test/resources/invalid_definitions.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+- name: Snowflake
```
Comment (malikdiarra) [inline airbyte-config/init/src/test/resources/invalid_definitions.yaml:1]: Was this intended to be part of that review? I do not see it being used anywhere here.

**Review by malikdiarra — APPROVED** [review]
Looks good. Having some tests on ApiPojoConverters could be useful.

File: airbyte-config/init/src/test/resources/invalid_definitions.yaml (hunk 1 above)
Comment (cgardens) [reply airbyte-config/init/src/test/resources/invalid_definitions.yaml:1]: nope. i was trying to write a test for `YamlSeedConfigPersistence` to check that the validation logic works, but unfortunately with the way it is structure that's really hard to do. forgot to remove this.

**Review by lmossman — APPROVED** [review]
LGTM

File: airbyte-config/init/src/main/resources/seed/destination_definitions.yaml [hunk 2]
```diff
@@ -188,6 +188,19 @@
   dockerImageTag: 0.4.15
   documentationUrl: https://docs.airbyte.io/integrations/destinations/snowflake
   icon: snowflake.svg
+  resourceRequirements:
+    default:
+      cpu_limit: "1.0"
+      cpu_request: "1.0"
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-config/init/src/main/resources/seed/destination_definitions.yaml:194]: hah. good question. i'll add a comment explaining.

**Pull request merged**